File: 2term/02.19/4calculator/4calculator/StackInMassive.cs
using System;
using ElementType = System.Int32;

namespace NStackInMassive
{
    // ..
    public class Stack
    {
        public Stack()
        {
            mas = new ElementType[size];
            numberElement = 0;
        }

        // add element newValue to tail.
        public void Push(ElementType newValue)
        {
            if (numberElement == size)
                return;
            mas[numberElement] = newValue;
            numberElement++;
        }

        // delete element in head and return value.
        public ElementType Pop()
        {
            if (numberElement == 0)
                return -1;
            return mas[--numberElement]; 
        }

        // print stack.
        public void Print()
        {
            if (numberElement == 0)
            {
                Console.WriteLine("Stack is empty");
                return;
            }
            for (int i = 0; i < numberElement - 1; i++)
                Console.Write("{0} -> ", mas[i]);
            Console.Write("{0}", mas[numberElement - 1]); 
            Console.WriteLine();
        }

        // return value of head.
        public ElementType ValueHead()
        {
            return mas[numberElement - 1];
        }

        private ElementType[] mas;
        // size array.
        private int size = 100;
        //number element in massive.
        private int numberElement;
    }
}

File: Qt/02.27/calculate/stackInMassive.cpp
#include "stackInMassive.h"
#include "stack.h"
using namespace stackNamespace;

StackInMassive::StackInMassive() : length(0), mas(new int[100])
{
//    mas = new int[100];
//    length = 0;
}

StackInMassive::~StackInMassive()
{
    delete mas;
}

// Add element value to begin stack.
void StackInMassive::push(int value)
{
    mas[length] = value;
    length++;
}

// Delete first element in stack and return its value.
int StackInMassive::pop()
{
    if (length <= 0)
        return -1;
    length--;
    return mas[length];
}

// return value in head.
int StackInMassive::head() const
{
    if (length <= 0)
        return -1;
    return mas[length - 1];
}

